File: hw/aymacana/u2/hw19Inheritance/FarmSystemJava/src/main/java/ec/edu/espe/farmsystem/model/Cage.java
package ec.edu.espe.farmsystem.model;

/**
 *
 * @author Mateo Aymacaña, T.A.P. (The Art of Programming), @ESPE
 */
public class Cage {

    private int id;
    private String descrption;
    private int type; //1 coop, 2 stables, 3 pens
    private Location location;

    @Override
    public String toString() {
        return "Cage{" + "id=" + id + ", descrption=" + descrption + ", type=" + type + ", location=" + location + '}';
    }

    public Cage(int id, String descrption, int type, Location location) {
        this.id = id;
        this.descrption = descrption;
        this.type = type;
        this.location = location;
    }

    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the descrption
     */
    public String getDescrption() {
        return descrption;
    }

    /**
     * @param descrption the descrption to set
     */
    public void setDescrption(String descrption) {
        this.descrption = descrption;
    }

    /**
     * @return the type
     */
    public int getType() {
        return type;
    }

    /**
     * @param type the type to set
     */
    public void setType(int type) {
        this.type = type;
    }

    /**
     * @return the location
     */
    public Location getLocation() {
        return location;
    }

    /**
     * @param location the location to set
     */
    public void setLocation(Location location) {
        this.location = location;
    }

}

File: hw/aymacana/u2/hw19Inheritance/FarmSystemPY/Cage.py
from Location import Location

class Cage:
    def __init__(self, id: int, description: str, type: int, location: Location):
        self._id = id
        self._description = description
        self._type = type  # 1 coop, 2 stables, 3 pens
        self._location = location
    
    # Getters
    @property
    def id(self) -> int:
        return self._id
    
    @property
    def description(self) -> str:
        return self._description
    
    @property
    def type(self) -> int:
        return self._type
    
    @property
    def location(self) -> Location:
        return self._location
    
    # Setters
    @id.setter
    def id(self, id: int) -> None:
        self._id = id
    
    @description.setter
    def description(self, description: str) -> None:
        self._description = description
    
    @type.setter
    def type(self, type: int) -> None:
        self._type = type
    
    @location.setter
    def location(self, location: Location) -> None:
        self._location = location
    
    def __str__(self) -> str:
        return (f"Cage{{id={self._id}, description='{self._description}', "
                f"type={self._type}, location={self._location}}}")